tests/h/activity/query_test.py

```diff
@@ -404,12 +404,8 @@ def test_it_returns_each_annotations_group(self, _fetch_groups, pyramid_request)
             for bucket in timeframe.document_buckets.values():
                 presented_annotations.extend(bucket.presented_annotations)
 
-        for group in _fetch_groups.return_value:
-            for presented_annotation in presented_annotations:
-                if presented_annotation["group"] == group:
-                    break
-            else:
-                assert False
+        expected_groups = [anno["group"] for anno in presented_annotations]
+        assert _fetch_groups.return_value == expected_groups
```

Comment:
This is both simpler, and also much more obvious about what is being asserted. At a larger scale I've no idea what this test file is up to.